p2p/pex/pex_reactor.go

```diff
@@ -312,7 +318,10 @@ func (r *PEXReactor) ReceiveAddrs(addrs []*p2p.NetAddress, src Peer) error {
 	}
 	r.requestsSent.Delete(id)
 
-	srcAddr := src.SocketAddr()
+	srcAddr, err := src.NodeInfo().NetAddress()
```

Comment:
Actually I'm not sure we need or want to do this here. We only care about the src IP, not port, so this fix isn't as relevant. Would be nice to formalize that in the type system somehow so it's clear.

This is fine as is, but it means we will do the DNS lookup for the source (if they have a name instead of just an IP) when we receive addresses from them, and if it fails, we'll ignore the addresses they sent us and we will disconnect that peer.

Reply:
Sounds like two distinct problems:

* only store src IP when receiving addresses
* use src.IP to reconnect and not DNS

Or do I miss-understand?

Reply:
> use src.IP to reconnect and not DNS

Where did you get this from ? Aren't we already doing this ?

Reply:
From your comment:

> This is fine as is, but it means we will do the DNS lookup for the source (if they have a name instead of just an IP)